src/main/java/cat/udl/eps/softarch/demo/handler/ShelterEventHandler.java

```diff
@@ -0,0 +1,29 @@
+package cat.udl.eps.softarch.demo.handler;
+
+import cat.udl.eps.softarch.demo.domain.Shelter;
+import cat.udl.eps.softarch.demo.repository.ShelterRepository;
+import org.springframework.data.rest.core.annotation.HandleBeforeCreate;
+import org.springframework.data.rest.core.annotation.RepositoryEventHandler;
+import org.springframework.stereotype.Component;
+
+@Component
+@RepositoryEventHandler
+public class ShelterEventHandler {
+    final ShelterRepository shelterRepository;
+
+    public ShelterEventHandler(ShelterRepository shelterRepository) {
+        this.shelterRepository = shelterRepository;
+    }
+
+    @HandleBeforeCreate
+    public void handleShelterPreCreate(Shelter shelter) {
+        if (shelter.getCreatedAt() == null) {
+            throw new IllegalArgumentException("Shelter must have a creation date");
```

Comment:
It is better, to set the dates here to make sure that we are not tricked by the API user to consider an unrealistic creation (or modification) date. Therefore, in the handler, before creation, set the created at and updated at times to ZonedDateTime.now(). Then, also use the before save handler to update the updated at time field before saving the entity

Reply:
The changes commented have been implemented , with an additional scenario